wp-config.php

```diff
@@ -47,7 +47,23 @@ function tribe_getenv( $name, $default = null ) {
 // Conditionally load tests-config if the proper header or user agent is present
 // ==============================================================
 
-if ( ( isset( $_SERVER['HTTP_X_TRIBE_TESTING'] ) || ( isset( $_SERVER['HTTP_USER_AGENT'] ) && $_SERVER['HTTP_USER_AGENT'] === 'tribe-tester' ) || tribe_getenv( 'WPBROWSER_HOST_REQUEST' ) ) && file_exists( __DIR__ . '/tests-config.php' ) ) {
+/**
+ * Check if current request is being made by Codeception.
+ *
+ * @return bool
+ */
+function tribe_is_codeception(): bool {
```

Comment:
During the tests bootstrap process, would there be a way to set a define of something like `define( 'TRIBE_TESTS', true );` or some other flag that could be read here to avoid running these checks on every request?

Reply:
Good point! These checks were already there, I just refactored them into something more readable.

I did some benchmarks, though, and improved it a bit by delaying the `file_exists` check only on test requests.

1.  Running a function that just `returns true` 1 million times: 1 second
2.  Running `tribe_is_codeception` 1 million times, before refactoring: 6.4 seconds
3.  Running `tribe_is_codeception` 1 million times, after postponing `file_exists` only on test requests: 3.6 seconds
4.  Running `tribe_is_codeception` 1 million times,  after postponing `file_exists` only on test requests and using `getenv` instead of `tribe_getenv`: 2.6 seconds

If we could safely use `getenv` there instead of `tribe_getenv` we would get timings that are a little better. For a single request, though, any of those timings are negligible, even if `tribe_is_codeception` is called multiple times.

I've pushed a commit with scenario number 3.